build.rs

```diff
@@ -1,73 +1,34 @@
-use std::io::Result;
+use prost_types::{FileDescriptorProto, FileDescriptorSet};
+use prost::Message;
+
+fn main() -> std::io::Result<()> {
+    let protos = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/protos.list"))
+        .lines()
+        .collect::<Vec<&str>>();
+
+    prost_build::Config::new()
+        // should probably switch to this
+        //.btree_map(&["."])
+        .out_dir("./out")
+        .compile_protos(protos.as_slice(), &["protos/"])?;
+
+    let apis = std::fs::read_to_string("./openapi/api-resources.json")?;
+
+    let buf = std::fs::read("./protos.fds").unwrap();
+    let fds = FileDescriptorSet::decode(&*buf).unwrap(); // pulls in proto::Message
+
+    // NB: FDS fields: https://github.com/tokio-rs/prost/blob/32bc87cd0b7301f6af1a338e9afd7717d0f42ca9/prost-types/src/protobuf.rs#L1-L7
+    // FDS usage: https://github.com/tokio-rs/prost/blob/32bc87cd0b7301f6af1a338e9afd7717d0f42ca9/prost-build/src/lib.rs#L765-L825
+    for f in fds.file {
+        use std::io::Write;
+        if let Some(pkg) = f.package {
+            let pkgpath = std::path::Path::new("./out").join(format!("{}.rs", pkg));
+            let generics = format!("// TODO genericsfor {}\n", pkg);
```

Comment:
> I think we can continue on from something like this and find corresponding types from the `api-resources.json` and inject as appropriate.

Yeah, we should be able to get all structs within this package from `f.message_type` vec.

Reply:
Have some time to start looking at this properly again now. Matching seems possible, but I might need to restructure the root keys in `api-resources.json`. The `apiGroupVersion` seems a little hard to match with `f.package` (and AFAIKT i don't have a better one in the `FileDescriptorSet` from a quick debug dump).